Add `/fix` slash command

dlqqq opened the pull request:
This PR introduces a new `/fix` slash command that fixes a code cell with error output when selected in an open notebook.

- `/fix` only works when a code cell with error output is the active cell. Otherwise the UI prevents `/fix` from being sent.
- `/fix` also accepts additional instructions in natural language following `/fix`.
    For example: `/fix Fix this code cell by using the numpy library only.`
- The code action toolbar can used to quickly replace the active cell with the revised code suggested by Jupyternaut.

## Demo video

https://github.com/jupyterlab/jupyter-ai/assets/44106031/87478ea8-969a-4c14-8068-b36b00acdcc4

srdas reviewed (commented):
Thanks for adding this, it is very useful, and should really help developers a lot. 

[1] Tested the branch and it works well and nothing else seems to be broken. 

[2] Tested that the `/fix` command works in an active cell that has an error whether you click on the code or the output. 

[3] Tested that when the cell does not have an error, then `/fix` cannot be used. 

[4] Tested that when the cell has only a warning, then `/fix` cannot be used. Do we want to have `/fix` work for cells with warnings?

[5] Nit: "Insert below active cell" (see below) does not say if the code will be inserted in the cell below (overwriting it) or insert a new cell and add the new code to it. The action is the latter but may be we want to point it out. (This is not directly related to this PR.)
![image](https://github.com/jupyterlab/jupyter-ai/assets/29005/5b040028-16b8-4f65-868d-08926f8c2408)

[6] I am getting a new chat user as shown below, is this intended?
![image](https://github.com/jupyterlab/jupyter-ai/assets/29005/9925e9d1-d03b-4e67-aa99-0d4f07e6095f)

dlqqq commented on packages/jupyter-ai/jupyter_ai/chat_handlers/default.py:
```diff
@@ -43,19 +43,6 @@ def create_llm_chain(
                 llm=llm, prompt=prompt_template, verbose=True, memory=self.memory
             )
 
-    def clear_memory(self):
```
This method does not belong in the default chat handler, and is simple enough to re-implement. Therefore I'm comfortable removing this unused method here to keep the code concise.

dlqqq commented on packages/jupyter-ai/jupyter_ai/chat_handlers/fix.py:
```diff
@@ -0,0 +1,90 @@
+)
+
+
+class FixChatHandler(BaseChatHandler):
```
Done in latest revision. 👍

dlqqq commented on packages/jupyter-ai/jupyter_ai/chat_handlers/fix.py:
```diff
@@ -0,0 +1,90 @@
+    async def process_message(self, message: HumanChatMessage):
+        if not (message.selection and message.selection.type == "cell-with-error"):
+            self.reply(
+                "`/fix` be given a cell with error output. Please click on a cell with error output and retry.",
```
Done in latest revision. 👍

dlqqq commented on packages/jupyter-ai/jupyter_ai/chat_handlers/fix.py:
```diff
@@ -0,0 +1,90 @@
+
+from .base import BaseChatHandler, SlashCommandRoutingType
+
+FIX_STRING_TEMPLATE = """
```
Thank you for calling this out! I've added user documentation in the latest revision. 

### Screenshots

<img width="746" alt="Screenshot 2024-06-11 at 5 05 02 PM" src="https://github.com/jupyterlab/jupyter-ai/assets/44106031/2f9d5afa-413d-4741-a566-8ead17693652">
<img width="757" alt="Screenshot 2024-06-11 at 5 05 10 PM" src="https://github.com/jupyterlab/jupyter-ai/assets/44106031/95002119-e336-4073-b935-e1e7d0d99965">

dlqqq commented:
@srdas Thank you for the quick review! I've addressed your review feedback in the threads above.

> [6] I am getting a new chat user as shown below, is this intended?

This is likely because you have `jupyter-collaboration` installed in your environment. When this is the case, Jupyter AI automatically infers your identity from the RTC identity provider. When RTC is enabled, the name shown in the chat panel matches the name shown in the collaborators panel.

srdas reviewed (approved):
Awesome work!! Verified all suggestions are implemented! TY!
